api/src/main/java/org/pmiops/workbench/actionaudit/auditors/AdminAuditorImpl.java

```diff
@@ -134,4 +134,34 @@ public void fireUnlockWorkspaceAction(long workspaceId) {
 
     actionAuditService.send(event);
   }
+
+  public void firePublishWorkspaceAction(long workspaceId, String category) {
+    auditPublishOrUnpublishAction(workspaceId, category, ActionType.PUBLISH);
+  }
+
+  public void fireUnPublishWorkspaceAction(long workspaceId, String category) {
+    auditPublishOrUnpublishAction(workspaceId, category, ActionType.UNPUBLISH);
+  }
+
+  void auditPublishOrUnpublishAction(long workspaceId, String category, ActionType action) {
+    DbUser dbUser = userProvider.get();
+    String actionId = actionIdProvider.get();
+    long timestamp = clock.millis();
+
+    ActionAuditEvent event =
+        ActionAuditEvent.builder()
+            .actionId(actionId)
+            .actionType(action)
+            .agentType(AgentType.ADMINISTRATOR)
+            .agentEmailMaybe(dbUser.getUsername())
+            .agentIdMaybe(dbUser.getUserId())
+            .targetType(TargetType.WORKSPACE)
+            .targetIdMaybe(workspaceId)
+            .targetPropertyMaybe("Category")
+            .newValueMaybe(category)
```

Comment:
I think we want this to be the **previous** category when unpublishing.  This is how it looks now, which I feel is a bit strange.

<img width="721" alt="Screenshot 2024-06-27 at 2 35 07 PM" src="https://github.com/all-of-us/workbench/assets/2701406/2f1b255a-5e97-4733-9eff-daa561a214e8">